wp-admin/js/customize-controls-site-icon-pwa.js

```diff
@@ -31,6 +31,13 @@ wp.customize(
 				// Update active state whenever the site_icon setting changes.
 				siteIconSetting.bind(updateActive);
 
+				// Change the site_icon_maskable if site_icon is not set.
+				siteIconSetting.bind((newSiteIconValue) => {
+					if (!newSiteIconValue) {
+						siteIconMaskableSetting(false);
+					}
+				});
+
```

Comment:
This does capture @pooja-muchandikar's comment in https://github.com/GoogleChromeLabs/pwa-wp/issues/304#issuecomment-1102267429:

> If Site icon was set and set as maskable and then removed the site icon, Maskable Icon setting remains checked.

However, I didn't think this was necessarily a bug. It's just whether one thinks of the maskable property being linked to the icon, or if maskable is something that you would persist across icon changes.

Reply:
> However, I didn't think this was necessarily a bug.

Agreed, rather it is just a check that make sure that by default we are assuming that an icon is maskable. 

If we had to save maskable property in an icon meta then we will be able to guess it correctly, for an icon maskable property should be true or not, but saving maskable property in meta could be overkill IMO.